lib/sentry/transaction.ex

```diff
@@ -0,0 +1,78 @@
+defmodule Sentry.Transaction do
+  @moduledoc """
+  The struct for the **transaction** interface.
+
+  See <https://develop.sentry.dev/sdk/event-payloads/transactions>.
+  """
+
+  @moduledoc since: "11.0.0"
+
+  alias Sentry.{Config, UUID, Interfaces.Span, Interfaces.SDK}
+
+  @typedoc since: "11.0.0"
+  @type t() ::
+          %__MODULE__{
+            # Required
+            event_id: <<_::256>>,
+            start_timestamp: String.t() | number(),
+            timestamp: String.t() | number(),
+            platform: :elixir,
+            contexts: %{
+              required(:trace) => %{
+                required(:trace_id) => String.t(),
+                required(:span_id) => String.t(),
+                optional(:parent_span_id) => String.t(),
+                optional(:op) => String.t(),
+                optional(:description) => String.t(),
+                optional(:status) => String.t()
+              }
+            },
+
+            # Optional
+            environment: String.t(),
+            transaction: String.t(),
+            transaction_info: map(),
+            measurements: map(),
+            type: String.t(),
+            tags: map(),
+            data: map(),
+
+            # Interfaces
+            spans: [Span.t()],
```

Comment:
A transaction without spans is ok?

Reply:
@whatyouhide yes! the only really required piece of span information is meant to go to context.trace which is why I defined it as a required map in the type spec. spans list is only needed when you have child spans, and that's not always the case.